File: backend/Controllers/MatchingController.cs
using Microsoft.AspNetCore.Mvc;
using PeerGrid.Backend.Models;
using PeerGrid.Backend.Services;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace PeerGrid.Backend.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class MatchingController : ControllerBase
    {
        private readonly MatchingService _matchingService;

        public MatchingController(MatchingService matchingService)
        {
            _matchingService = matchingService;
        }

        [HttpGet("direct")]
        public async Task<ActionResult<List<User>>> FindMatches([FromQuery] string skillNeeded, [FromQuery] int requesterId)
        {
            var matches = await _matchingService.FindMatchesWithScoreAsync(skillNeeded, requesterId);
            return Ok(matches);
        }

        [HttpGet("triangular")]
        public async Task<ActionResult<List<User>>> FindTriangularMatch([FromQuery] int userId)
        {
            var match = await _matchingService.FindTriangularMatchAsync(userId);
            if (match == null || match.Count == 0)
            {
                return NotFound(new { message = "No triangular match found" });
            }
            return Ok(match);
        }
    }
}

File: backend(Spring Boot)/src/main/java/com/peergrid/backend/controller/MatchingController.java
package com.peergrid.backend.controller;

import com.peergrid.backend.entity.User;
import com.peergrid.backend.service.MatchingService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/matching")
@CrossOrigin(origins = "*")
public class MatchingController {

    @Autowired
    private MatchingService matchingService;

    @GetMapping("/direct")
    public ResponseEntity<List<User>> findMatches(@RequestParam String skillNeeded, @RequestParam Integer requesterId) {
        List<User> matches = matchingService.findMatchesWithScore(skillNeeded, requesterId);
        return ResponseEntity.ok(matches);
    }

    @GetMapping("/triangular")
    public ResponseEntity<?> findTriangularMatch(@RequestParam Integer userId) {
        List<User> match = matchingService.findTriangularMatch(userId);
        if (match == null || match.isEmpty()) {
            return ResponseEntity.status(404).body("{\"message\": \"No triangular match found\"}");
        }
        return ResponseEntity.ok(match);
    }
}
